src/conways.rs

```diff
@@ -17,10 +19,176 @@ impl ConwaysGrid {
         Self { grid }
     }
 
+    fn modify_cell(&mut self, pos: Position, new_state: CellState) {
+        if let Some(grid_row) = self.grid.get_mut(pos.0) {
+            if let Some(cell) = grid_row.get_mut(pos.1) {
+                *cell = new_state;
+            }
+        }
+    }
+
+    pub fn from_alive_cells(alive_cells: Vec<Position>) -> Self {
+        let mut conways_grid = Self::default();
+        alive_cells
+            .into_iter()
+            .for_each(|cell| conways_grid.modify_cell(cell, CellState::Alive));
+        conways_grid
+    }
+
     pub fn next_iteration(&mut self) {
-        for row in self.grid.iter() {
-            for cell in row.iter() {
-                todo!("Implement game rules");
+        let previous_grid = self.grid.clone();
+        for row in 0..ROWS {
+            for col in 0..COLS {
+                self.compute_next_state(&previous_grid, (row, col));
+            }
+        }
+    }
+
+    fn compute_next_state(&mut self, previous_grid: &[Vec<CellState>], pos: Position) {
+        let neighbors: Vec<Position> = self.get_neighbor_position(pos);
+        let alive = Self::get_alive_count(previous_grid, neighbors);
+        let (row, col) = pos;
+        if let Some(row) = self.grid.get(row) {
+            if let Some(cell) = row.get(col) {
+                match cell {
+                    CellState::Alive => {
+                        if !(2..=3).contains(&alive) {
+                            self.modify_cell(pos, CellState::Dead);
+                        }
+                    }
+                    CellState::Dead => {
+                        if alive == 3 {
+                            self.modify_cell(pos, CellState::Alive);
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    fn get_neighbor_position(&self, pos: Position) -> Vec<Position> {
```

Comment:
You can directly do (row, col): Position here and avoid destructuring in the following line